WASdev/azure.liberty.aks
# [MVP release] Deploy a Java application with Open Liberty/WebSphere Liberty on an AKS cluster

## Description
This PR is to resolve the following issues:
* Fix WASdev/azure.liberty.aro/issues/16
* Fix WASdev/azure.liberty.aro/issues/17
* Fix WASdev/azure.liberty.aro/issues/19

## Change summary
* Support to create an AKS cluster or reuse an existing one 
* Support to create an ACR instance or reuse an existing one 
* Use deployment script to orchestrate the whole deployment process 
   * Install Open Liberty Operator
   * Uploading user specified application package (.war) is optional
   * Support to choose Open Liberty/WebSphere Liberty base image
   * Support to choose Java 8/Java 11 runtime
   * Build application image or use an Open Liberty/WebSphere Liberty base image
   * Deploy Open Liberty Application to the cluster
* Update readme

## Marketplace offer for testing
To facilitate PR review, I updated the testing offer with the latest code changes included in the PR, so reviewers can try by themselves to test, verify and share their feedback:
* Open the [preview link](https://portal.azure.com/#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewazure-liberty-aks) in Azure portal
   * If you see **Could not create the marketplace item**, click your profile at the top-right of the page > Switch directory > Select "WAS-MSFT-TEAM"
* Click **Create** to start the offer 

## Test cases
The following test cases have already been verified before sending out the PR:
* Create a new AKS cluster, user uploaded app, Open Liberty base image, Java 8 runtime;
* Use an existing AKS cluster, user uploaded app, Open Liberty base image, Java 8 runtime;
* Use an existing AKS cluster, user uploaded app, WebSphere Liberty base image, Java 11 runtime;
* Use an existing AKS cluster, no user app, Open Liberty base image, Java 8 runtime;
* Use an existing AKS cluster, no user app app, WebSphere Liberty base image, Java 11 runtime;
* The pipeline can successfully run and generate a zip file for offer update, see [completed builds](https://github.com/majguo/azure.liberty.aks/actions) 

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>
